packages/storage/src/append_store.rs

```diff
@@ -162,16 +226,42 @@ impl<'a, T: Serialize + DeserializeOwned, Ser: Serde> AppendStore<'a, T, Ser> {
         let len = self.get_len(storage)?;
 
         if pos >= len {
-            return Err(StdError::generic_err("DequeStorage access out of bounds"));
+            return Err(StdError::generic_err("AppendStore access out of bounds"));
         }
```

Comment:
Yes,  page number requirements are checked in match statement. Thus it should be impossible for us to reach the match arm `std::cmp::Ordering::Greater => {Err(StdError::generic_err("AppendStore access out of bounds"))}`. However, even then the match statement isn't checking the full index requirement that `pos < len` as it is possible for `pos` to be in the same page as `len` and yet still be `pos >= len`

Reply:
Hmm correct.
So in that case I think instead of the `match` it's better to do:
```rust
if pos_page < max_page {
    ...
} else {
    ...
}
```